src/BaseBinaryStar.cpp

```diff
@@ -1390,366 +1390,388 @@ void BaseBinaryStar::ResolveCoalescence() {
-double BaseBinaryStar::CalculateSemiMajorAxisPostSupernova(const double p_KickVelocity,
-                                                           const double p_TotalMassPreSN,
-                                                           const double p_TotalMassPostSN,
-                                                           const double p_KickTheta,
-                                                           const double p_KickPhi) {
+    // Set relevant preSN parameters 
+    m_EccentricityPreSN = m_EccentricityPrev;                                                 
+    m_SemiMajorAxisPreSN = m_SemiMajorAxisPrev;                                               
 
-    double r_2           = 2.0 / m_Radius;
-    double quadraticTerm = 1.0 + (2.0 * p_KickVelocity * cos(p_KickTheta) * cos(p_KickPhi)) + (p_KickVelocity * p_KickVelocity);
+    double totalMassPreSN = m_Supernova->MassPrev() + m_Companion->MassPrev();                                           // Total Mass preSN
+    double reducedMassPreSN = m_Supernova->MassPrev() * m_Companion->MassPrev() / totalMassPreSN;                        // Reduced Mass preSN
+    m_Supernova->SetOrbitalEnergyPreSN(CalculateOrbitalEnergy(reducedMassPreSN, totalMassPreSN, m_SemiMajorAxisPreSN));  // Orbital energy preSN
 
-    return 1.0 / (r_2 - ((p_TotalMassPreSN / p_TotalMassPostSN) * (r_2 - (1.0 / m_SemiMajorAxisPrime)) * quadraticTerm));
-}
+    // Define the natal kick vector (see above for precise definitions of the angles)
+    double theta = m_Supernova->SN_Theta();         // Angle out of the binary plane
+    double phi   = m_Supernova->SN_Phi();           // Angle in the binary plane
+    Vector3d natalKickVector = m_Supernova->SN_KickVelocity() * Vector3d(cos(theta)*cos(phi), 
+                                                                         cos(theta)*sin(phi),
+                                                                         sin(theta));
 
+    // Check if the system is already unbound
+    if (IsUnbound()) {                                                                                    // Is system already unbound?
 
-/*
- * Resolves supernova event - one of the stars has gone supernova!
- *
- * Assign a random supernova kick according to the user specified options and then update the orbit
- *
- * JR: todo: flesh-out this documentation
- *
- *
- * bool ResolveSupernova()
- *
- * @return                                      True if a supernova event occurred, otherwise false
- */
-bool BaseBinaryStar::ResolveSupernova() {
+        m_Supernova->UpdateComponentVelocity( natalKickVector.RotateVector(m_ThetaE, m_PhiE, m_PsiE));    // yes - only need to update the velocity of the star undergoing SN
 
-    if (!m_Supernova->IsSNevent()) return false;                                                                                    // not a supernova event - bail out (or bale out depending whence you hail...) passively
+        // The quantities below are mostly meaningless 
```

Comment:
I'm not sure what legacy would choose for these, but this section sets binary parameters for already disrupted binaries, so they don't mean much physically. Orbital velocity (pre and post) and dimensionless kick don't have a meaning, so I set them to NaN, but I keep the e>1 and a<0 values since it's clear by eye that these apply to disrupted binaries. 

I doubt this will impact anything else, but if it does, I'd rather it throw an error so we can properly debug it.